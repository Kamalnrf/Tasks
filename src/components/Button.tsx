import React from 'react'
import {
  AccessibilityProps,
  StyleProp,
  StyleSheet,
  ViewStyle,
  TouchableOpacity,
  Text,
  TextStyle,
} from 'react-native'
import {colors} from '../constants'

type Props = AccessibilityProps & {
  children: string
  onPress: () => void
  containerStyle?: StyleProp<ViewStyle>
  textStyle?: StyleProp<TextStyle>
  disabled: boolean
}

const styles = StyleSheet.create({
  txt: {
    fontFamily: 'Roboto_400Regular',
    lineHeight: 46,
    fontSize: 15,
    color: colors.blue_92A9EC,
    alignSelf: 'flex-end',
    paddingRight: 10,
  },
})

const Button = ({
  containerStyle,
  children,
  onPress,
  textStyle,
  disabled,
  ...props
}: Props) => (
  <TouchableOpacity
    style={containerStyle}
    onPress={onPress}
    accessibilityRole="button"
    disabled={disabled}
    {...props}
  >
    <Text
      style={[
        styles.txt,
        textStyle,
        disabled ? {color: colors.gray_9C9DA2} : {},
      ]}
    >
      {children}
    </Text>
  </TouchableOpacity>
)

export default Button
