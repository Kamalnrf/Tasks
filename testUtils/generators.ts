import faker from 'faker'
import {Tag, Task} from '../src/types'

function generateList<T>(fn: (override?: {}) => T) {
  return (n: number): Array<T> =>
    new Array(n).fill('').reduce((acc, _) => [...acc, fn()])
}

export const generateTag = (override = {}): Tag => ({
  id: faker.random.uuid(),
  name: faker.name.title(),
  ...override,
})

export const generateTags = generateList<Tag>(generateTag)

export const generateTask = (override = {}): Task => ({
  id: faker.random.uuid(),
  endTime: faker.date.recent().toISOString(),
  startTime: faker.date.past().toISOString(),
  tags: generateTags(10),
  title: faker.name.title(),
  ...override,
})
