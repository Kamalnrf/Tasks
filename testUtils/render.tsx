import React from 'react'
import {render as rtlRender} from '@testing-library/react-native'
import {QueryClient, QueryClientProvider} from 'react-query'
import {SafeAreaProvider} from 'react-native-safe-area-context'

const queryClient = new QueryClient()

function render(ui: JSX.Element, {...renderOptions} = {}) {
  function Wrapper({children}: {children: JSX.Element}) {
    return (
      <QueryClientProvider client={queryClient}>
        <SafeAreaProvider
          initialMetrics={{
            frame: {x: 0, y: 0, width: 0, height: 0},
            insets: {top: 0, left: 0, right: 0, bottom: 0},
          }}
        >
          {children}
        </SafeAreaProvider>
      </QueryClientProvider>
    )
  }
  return rtlRender(ui, {wrapper: Wrapper, ...renderOptions})
}

export * from '@testing-library/react-native'

export {render}
