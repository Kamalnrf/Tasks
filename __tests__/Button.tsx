import React from 'react'
import Button from '../src/components/Button'
import {fireEvent, render} from '../testUtils/render'

describe('Test Button', () => {
  it('should render button with given text and to be tapable', () => {
    const onPress = jest.fn()
    const {toJSON, queryByA11yRole} = render(
      <Button onPress={onPress}>Test Button</Button>,
    )

    fireEvent.press(queryByA11yRole('button'))
    expect(onPress).toHaveBeenCalledWith()
    expect(toJSON()).toMatchSnapshot()
  })
})
