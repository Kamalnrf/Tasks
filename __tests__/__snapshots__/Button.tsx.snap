// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button should render button with given text and to be tapable 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "flex-end",
            "color": "#92A9EC",
            "fontFamily": "Roboto_400Regular",
            "fontSize": 15,
            "lineHeight": 46,
            "paddingRight": 10,
          },
          undefined,
        ]
      }
    >
      Test Button
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
